casper/src/main/scala/coop/rchain/casper/blocks/proposer/BlockCreator.scala

```diff
@@ -1,228 +1,109 @@
-                preState.preStateHash
-              )
-              for {
-                slashingDeploys <- prepareSlashingDeploys(ilmFromBonded, rand, deploys.size)
-                // make sure closeBlock is the last system Deploy
-                systemDeploys = slashingDeploys :+ CloseBlockDeploy(
-                  rand.splitByte((deploys.size + slashingDeploys.size).toByte)
-                )
-                preStateHash = preState.preStateHash.toByteString
-                checkpointData <- InterpreterUtil.computeDeploysCheckpoint(
-                                   deploys.toSeq,
-                                   systemDeploys,
-                                   rand,
-                                   blockData,
-                                   preStateHash
-                                 )
-                (postStateHash, processedDeploys, processedSystemDeploys) = checkpointData
+      BlockDagStorage[F].pooledDeploys
+        .map(_.filterKeys(deploys.toSet).values.toSeq)
+        .flatMap(
+          InterpreterUtil.computeDeploysCheckpoint[F](
+            _,
+            slashDeploys :+ closeDeploy,
+            rand,
+            blockData,
+            preStateHash.toByteString
+          )
+        )
+    }
 
-                _ <- Span[F].mark("before-packing-block")
+    /** Create attestation. */
+    def attest: F[StateTransitionResult] = Sync[F].delay {
+      val postStateHash          = preStateHash.toByteString
+      val processedDeploys       = Seq()
+      val processedSystemDeploys = Seq()
+      (postStateHash, processedDeploys, processedSystemDeploys)
+    }
 
-                // Create block and calculate block hash
-                unsignedBlock = packageBlock(
-                  validatorIdentity.publicKey,
-                  blockData,
-                  justifications,
-                  preStateHash,
-                  postStateHash,
-                  processedDeploys,
-                  processedSystemDeploys,
-                  shardId,
-                  BlockVersion.Current,
-                  // Bonds data in the block is referring to finalized fringe
-                  //  not bonds in conflict scope
-                  preState.fringeBondsMap,
-                  // Rejected data in the block is referring to finalized fringe
-                  //  not rejections in conflict scope
-                  preState.fringeRejectedDeploys
-                )
-                _ <- Span[F].mark("block-created")
+    val postState =
+      if (shouldPropose) propose.map(_.some)
+      else (!suppressAttestation).guard[Option].traverse(_ => attest)
 
-                // Sign a block (hash should not be changed)
-                signedBlock = validatorIdentity.signBlock(unsignedBlock)
-                _           <- Span[F].mark("block-signed")
+    postState.map {
+      case None                                                            => BlockCreatorResult.noNewDeploys
```

Comment:
Block creator status should be removed, see comments above.

Reply:
How do you want to signal about the result of a propose? What should be the result? Are you thinking to store it in Proposer state?

Reply:
Checking the condition to propose or not should not be inside of the function doing the main logic. So if conditions are not satisfied propose should not be called.

We can leave it as is in this PR and fix it later.